src/tcache.c

```diff
@@ -422,32 +434,56 @@ tcache_stats_merge(tsdn_t *tsdn, tcache_t *tcache, arena_t *arena) {
 	}
 }
 
-bool
-tcaches_create(tsd_t *tsd, unsigned *r_ind) {
-	arena_t *arena;
-	tcache_t *tcache;
-	tcaches_t *elm;
+static bool
+tcaches_create_prep(tsd_t *tsd) {
+	bool err;
+
+	malloc_mutex_lock(tsd_tsdn(tsd), &tcaches_mtx);
 
 	if (tcaches == NULL) {
```

Comment:
The reason this is lazy is that we don't ever call ```tcaches_create()``` unless the application uses the "tcache.create" mallctl.  Automatic per thread tcaches don't use this array.